ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/GOSymbols/GOSymbols.cpp

```diff
@@ -103,7 +103,7 @@ void GOSymbols::addBuoyPoints(GraphicsOverlay* graphicsOverlay)
 
   // create the symbology for the points
   constexpr float size = 10.0f;
-  SimpleMarkerSymbol* sms = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, QColor("red"), size, this);
+  SimpleMarkerSymbol* sms = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, QColor("red"), size, this); // leak?
```

Comment:
That is a crazy warning from clazy. I guess clazy suppose that the range-for loop can have 0 loop, which is not true as the containers is const. I think this is a bug in clazy.

Reply:
I think this is a mistake of clang-tidy. The pattern is:
```cpp
p = new...;
for (...) {
  // use p
}
```
In this case, clang-tidy think the loop can be empty and then `p` is not used. And then there is a leak.

In our code, the loop runs over a const non-empty container and there is no leak. But clang-tidy doesn't know the loop is never empty.

We can ignore the warning in this case.